src/App.tsx

```diff
@@ -1,14 +1,83 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 import { Loader } from './components/Loader';
+import { TodoModal } from './components/TodoModal';
+
+function getPreparedTodos(todos: Todo[], query: string) {
+  let preparedTodos = [...todos];
+
+  if (query) {
+    const prepQuery = query.trim().toLowerCase();
+
+    preparedTodos = preparedTodos.filter(
+      todo => todo.title.toLowerCase().includes(prepQuery),
+    );
+  }
+
+  return preparedTodos;
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const visibleTodos = getPreparedTodos(todos, query);
+  const [originalTodos, setOriginalTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [modalVisible, setModalVisible] = useState(true);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  const handleOpenModal = (todo: Todo) => {
+    setModalVisible(true);
+    setSelectedTodo(todo);
+  };
+
+  const handleCloseModal = () => {
+    setModalVisible(false);
+    setSelectedTodo(null);
+  };
+
+  const handleQueryChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(e.target.value);
+  };
+
+  const handleQueryDelete = () => {
+    setQuery('');
+  };
+
+  const handleFilterCompleted = () => {
+    const filteredTodos = originalTodos.filter((todo) => todo.completed === true);
+
+    setTodos(filteredTodos);
+  };
+
+  const handleFilterActive = () => {
+    const filteredTodos = originalTodos.filter((todo) => todo.completed === false);
+
+    setTodos(filteredTodos);
+  };
+
+  const handleFilterAll = () => {
+    setTodos(originalTodos);
+  };
```

Comment:
it looks too complex, we can create the filterValue state here and pass the setFilterValue function to TodoFilter instead of these three functions (just imagine how much we will have to change if we suddenly need to add a few more select options)

and depending on the filterValue filter the array of todos